expertise/models/multifacet_recommender/specter.py

```diff
@@ -154,14 +177,15 @@ def set_archives_dataset(self, archives_dataset):
                 self.pub_author_ids_to_note_id[profile_id].append(publication['id'])
                 self.pub_note_id_to_title[publication['id']] = publication['content'].get('title', "")
                 self.pub_note_id_to_abstract[publication['id']] = publication['content'].get('abstract', "")
-                if publication['id'] in output_dict:
-                    output_dict[publication['id']]["authors"].append(profile_id)
-                else:
-                    paper_ids_list.append(publication['id'])
-                    output_dict[publication['id']] = {"title": self.pub_note_id_to_title[publication['id']],
-                                                      "abstract": self.pub_note_id_to_abstract[publication['id']],
-                                                      "paper_id": publication["id"],
-                                                      "authors": [profile_id]}
+                if not self.redis.exists(publication['id']):
```

Comment:
what happens if the publication is in redis?

Reply:
Nothing, we don't need to calculate embedding for this paper in that case, and specter will get these embeddings when running the `all_scores` method.